### PR TITLE
samples: openthread: FLASH/RAM optimization

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
Add configuration overlay with minimal configuration for Thread CLI singleprotocol and multiprotocol sample.

### CODE REVIEW

**Review by edmont — APPROVED** [review]

File: samples/openthread/cli/harness/overlay-cert.conf [hunk 1]
```diff
@@ -30,7 +30,6 @@ CONFIG_BT_LL_SOFTDEVICE_DEFAULT=y
 
 # Enable and configure Bluetooth LE
 CONFIG_BT=y
-CONFIG_BT_SMP=y
```
Comment (pszkotak) [inline samples/openthread/cli/harness/overlay-cert.conf:33]: Is the change in the certification overlay intentional?

Comment (lmaciejonczyk) [reply samples/openthread/cli/harness/overlay-cert.conf:33]: Yes, the certification tests do not create the bluetooth connection, only advertise. This option enables support for bluetooth secure connection.

**Review by pszkotak — APPROVED** [review]

**Review by ru-fu — CHANGES_REQUESTED** [review]
I added two commit with some small fixes. Please check and squash if okay!

We could move the commit that adds information to ug_app_dev.rst into a separate PR if you want. But would be nice to have. :)

**Comment by lmaciejonczyk** [comment]
> I added two commit with some small fixes. Please check and squash if okay!
> 
> We could move the commit that adds information to ug_app_dev.rst into a separate PR if you want. But would be nice to have. :)

It looks good to me. I've added missing parts. Review last commit please.

**Review by ru-fu — COMMENTED** [review]
Another small commit. :) 
Also, two comments I made yesterday but forgot to submit ... 🙄

File: doc/nrf/ug_app_dev.rst [hunk 2]
```diff
@@ -115,3 +117,46 @@ Complete the following actions to optimize the Bluetooth part of your applicatio
...
+* Disable features that your application does not use.
+  For example, disable the following features:
+
+  * Logging
+  * Network shell
+
+* Reduce the stack sizes of the Thread internal threads where possible.
+
+  Use the :ref:`zephyr:thread_analyzer` to analyze the stack usage.
+  You should enable the following configuration options to output the current use:
```
[the remark this answers is not in the document]
Comment (lmaciejonczyk) [reply doc/nrf/ug_app_dev.rst:138]: It is general, and already described here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/debugging/thread-analyzer.html#configuration

[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/nrf/ug_app_dev.rst:138]: And we have a link, so no need to duplicate. I'll remove. :)

File: doc/nrf/ug_app_dev.rst [hunk 3]
```diff
@@ -115,3 +117,46 @@ Complete the following actions to optimize the Bluetooth part of your applicatio
...
+  * :option:`CONFIG_THREAD_ANALYZER_USE_PRINTK`
+  * :option:`CONFIG_CONSOLE`
+  * :option:`CONFIG_UART_CONSOLE`
+  * :option:`CONFIG_SERIAL`
+  * :option:`CONFIG_PRINTK`
+
+  The following configuration options affect the stack sizes of the Thread threads:
+
+  * :option:`CONFIG_OPENTHREAD_THREAD_STACK_SIZE`
+  * :option:`CONFIG_NET_CONNECTION_MANAGER_STACK_SIZE`
```
[the remark this answers is not in the document]
Comment (lmaciejonczyk) [reply doc/nrf/ug_app_dev.rst:153]: It makes sense to me.

**Review by ru-fu — APPROVED** [review]
Looks good!

**Review by MarekPorwisz — APPROVED** [review]

**Comment by lmaciejonczyk** [comment]
@carlescufi PR is green, merge it please.

**Comment by lmaciejonczyk** [comment]
@rlubos Can you help to merge it please?

**Review by rlubos — APPROVED** [review]

**Pull request merged**